src/datastore/sync_methods/inject.js

```diff
@@ -313,12 +313,10 @@ module.exports = function inject (resourceName, attrs, options) {
     definition.emit('DS.beforeInject', definition, attrs)
   }
 
-  // start the recursive injection of data
+  // start the recursive injection of data; this will also call
+  // `definition.handleChange` for each item
   injected = _inject.call(_this, definition, resource, attrs, options)
 
-  // collection was modified
```

Comment:
The line I removed was doing essentially the same thing as what the batched notification does now. It would end up being duplicated functionality and data.